module/char/charpick.c

```diff
@@ -110,6 +110,21 @@ void ChooseCharacter(Character *characters, WORD num_characters, char *motd,
    }
 }
 /********************************************************************/
+/*
+ * compareCharacters:  Sorting alphabetically with new character slots always at the end.
+ */
+bool compareCharacters(const Character& a, const Character& b) {
+    // Place characters with flags != 1 (new users) at the end
+    if (a.flags != 1 && b.flags == 1) {
+        return true;
+    }
+    if (a.flags == 1 && b.flags != 1) {
+        return false;
+    }
+    // Neither character is new so compare names case-insensitively

```

Comment:
I don't think this is right.  We get here if both flags are 0, or both flags are 1.  Both characters could be new.